src/sync/atomic/asm_wasm.s

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include "textflag.h"
+
+TEXT ·SwapInt32(SB),NOSPLIT,$0-20
+	JMP ·SwapUint32(SB)
+
+TEXT ·SwapUint32(SB),NOSPLIT,$0-20
+	MOVD addr+0(FP), I0
+	MOVW (I0), old+16(FP)
+	MOVW new+8(FP), (I0)
+	RET
+
+TEXT ·SwapInt64(SB),NOSPLIT,$0-24
+	JMP ·SwapUint64(SB)
+
+TEXT ·SwapUint64(SB),NOSPLIT,$0-24
+	MOVD addr+0(FP), I0
+	MOVD (I0), old+16(FP)
+	MOVD new+8(FP), (I0)
+	RET
+
+TEXT ·SwapUintptr(SB),NOSPLIT,$0-24
+	JMP ·SwapUint64(SB)
```

Comment:
I don't know if it matters much, but the order of some of the funcs is not consistent.

Sometimes `Uintpr` variant comes after `Int64`, `Uint64` variants.

Other times `Uintptr` comes first, followed by `Int64` and `Uint64`.

It might be better to reorder them to be consistent.

Reply:
It was the same order as in asm_amd64.s, but you are right, it is quite inconsistent. I've reordered them.